### PR TITLE
Refactored FindView and related components

### LINKED ISSUE #531: Refactor FindView
### Description
Refactor FindView.vue and related components because :star: future-proofing :star:!

### Acceptance Criteria
- [x] Findview.vue is refactored
- [x] BookListComponent is refactored
- [x] LoanListComponent is refactored
- [ ] UserListComponent is refactored

### PR DESCRIPTION
**Pull request by te4-alexander-kjellberg**
## ⚙️ What issue did you implement or fix?
Fixes #531 

## 📜 Summary
- Refactored FindView.vue
- Refactored BookListComponent.vue

## 📝 Type of change

- [x] ✨ New feature or enhancement (non-breaking change which adds functionality)

## ✅ Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## 💩 How can this implementation be improved?
- Add tests and create types.

### CODE REVIEW

File: frontend/src/components/BookListComponent.vue [hunk 1]
```diff
@@ -15,28 +15,30 @@
...
 
-@Component({
+export default defineComponent({
+  name: 'BookListComponent',
   components: {
     ListComponent
+  },
+  setup() {
+    const booksModule: VuexModule = BooksModule;
+    const headers = [
```
Comment (te4-alexander-kjellberg1) [inline frontend/src/components/BookListComponent.vue:30]: Why not write it like this? This matches line 29 more.    

 const headers : object[] = [
      { text: 'Titel', value: 'title.name' },
      { text: 'Skick', value: 'condition' },
      { text: 'Sträckkod', value: 'barcode' },
 ];

File: frontend/src/components/BookListComponent.vue [hunk 2]
```diff
@@ -15,28 +15,30 @@
...
+      { text: 'Titel', value: 'title.name' },
+      { text: 'Skick', value: 'condition' },
+      { text: 'Sträckkod', value: 'barcode' },
```
Comment (te4-jonathan-ronsten) [inline frontend/src/components/BookListComponent.vue:33]: Change to 'Streckkod' Otherwise looks good

File: frontend/src/views/FindView.vue [hunk 3]
```diff
@@ -71,24 +70,33 @@ import UsersModule from '@/store/modules/UsersModule';
...
   },
-})
-export default class FindView extends Vue {
-  private RoleChecker: RoleChecker = RoleChecker;
-  users:boolean = false;
-  loans:boolean = false;
-  books:boolean = false;
+  setup(_: object, { root }: SetupContext): object {
+    const rolechecker = RoleChecker;
+    const users = ref(false as boolean);
```
Comment (te4-alexander-kjellberg1) [inline frontend/src/views/FindView.vue:82]: Why use "false as boolean" and not "false : boolean"

**Review by te4-jonathan-ronsten — CHANGES_REQUESTED** [review]
Looks good overall, check comment

File: frontend/src/views/FindView.vue (hunk 3 above)
Comment (te4-alexander-kjellberg) [reply frontend/src/views/FindView.vue:82]: It's the pro way

File: frontend/src/components/BookListComponent.vue (hunk 1 above)
Comment (te4-alexander-kjellberg) [reply frontend/src/components/BookListComponent.vue:30]: It's the pro way!!

**Review by te4-alexander-kjellberg1 — APPROVED** [review]
Nice

**Review by te4-jonathan-ronsten — APPROVED** [review]
Looks good now! Nice job

**Review by te4-david-jensen — CHANGES_REQUESTED** [review]
Please clean up that commit history before merging

**Pull request closed without merging**